Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Search.hs

```diff
@@ -193,8 +193,9 @@ handler merchant sReq = do
     listVehicleVariantHelper farePolicy = catMaybes $ everyPossibleVariant <&> \var -> find ((== var) . (.vehicleVariant)) farePolicy
 
     buildEstimates farePolicies result fromLocation toLocation specialLocationTag area routeInfo = do
-      driverPoolCfg <- getDriverPoolConfig merchant.id result.distance
-      estimateInfos <- buildEstimatesInfos fromLocation toLocation driverPoolCfg result farePolicies specialLocationTag area routeInfo
+      driverPoolCfgs <- getDriverPoolConfigs merchant.id result.distance
+      let estimateInfo = mapM (\cfg -> buildEstimatesInfos cfg.vehicleVariant fromLocation toLocation cfg result farePolicies specialLocationTag area routeInfo) driverPoolCfgs
```

Comment:
Here for each vehicle, you will create driver pool separately from db. Instead of that, create driver pool at once and then segregate them on basis of vehicle variant. Else this API will become very slow.